histomicstk/segmentation/label/isbfcpp.cpp

```diff
@@ -0,0 +1,344 @@
+      matrix180[i] = new int[ncols];
+    }
+    matrix270 = new int*[ncols];
+    for (int i = 0; i < ncols; i++) {
+      matrix270[i] = new int[nrows];
+    }
+
+    // rotate Matrix for 90, 180, 270 degrees
+    rotateMatrix(nrows, ncols, matrix00, matrix270);
+    rotateMatrix(ncols, nrows, matrix270, matrix180);
+    rotateMatrix(nrows, ncols, matrix180, matrix90);
+
+    // set defalut direction
+    int DX = 1;
+    int DY = 0;
+    // set the number of rows and cols for ISBF
+    int rowISBF = 3;
+    int colISBF = 2;
+
+    float angle;
+    // set size of X: the size of X is equal to the size of Y
+    int sizeofX;
+    // loop until true
+    while(1) {
+
+      int x;
+      int y;
+
+      x = curr->val_x;
+      y = curr->val_y;
+
+      int **h = new int*[rowISBF];
+      for (int i = 0; i < rowISBF; i++) {
+        h[i] = new int[colISBF];
+      }
+
+      // initialize a and b which are indices of ISBF
+      int a = 0;
+      int b = 0;
+
+      if((DX == 1)&&(DY == 0)){
+        for (int i = ncols-x-2; i < ncols-x+1; i++) {
+          for (int j = y-1; j < y+1; j++) {
+              h[a][b] = matrix90[i][j];
+              b++;
+          }
+          b = 0;
+          a++;
+        }
+        angle = M_PI/2;
+      }
+
+      else if((DX == 0)&&(DY == -1)){
+        for (int i = y-1; i < y+2; i++) {
+          for (int j = x-1; j < x+1; j++) {
+              h[a][b] = matrix00[i][j];
+              b++;
+          }
+          b = 0;
+          a++;
+        }
+        angle = 0;
+      }
+
+      else if((DX == -1)&&(DY == 0)){
+        for (int i = x-1; i < x+2; i++) {
+          for (int j = nrows-y-2; j < nrows-y; j++) {
+              h[a][b] = matrix270[i][j];
+              b++;
+          }
+          b = 0;
+          a++;
+        }
+        angle = 3*M_PI/2;
+      }
+
+      else{
+        for (int i = nrows-y-2; i < nrows-y+1; i++) {
+          for (int j = ncols-x-2; j < ncols-x; j++) {
+              h[a][b] = matrix180[i][j];
+              b++;
+          }
+          b = 0;
+          a++;
+        }
+        angle = M_PI;
+      }
+
+      // initialize cX and cY
+      // cX and cY indicate directions for each ISBF
+      vector<int> cX(1);
+      vector<int> cY(1);
+      cX.at(0)=0;
+      cY.at(0)=0;
+
+      if (h[1][0] == 1) {
+          // 'left' neighbor
+          cX.at(0) = -1;
+          cY.at(0) = 0;
+          DX = -1;
+          DY = 0;
+      }
+      else{
+          if((h[2][0] == 1)&&(h[2][1] != 1)){
+              // inner-outer corner at left-rear
+              cX.at(0) = -1;
+              cY.at(0) = 1;
+              DX = 0;
+              DY = 1;
+          }
+          else{
+              if(h[0][0] == 1){
+                  if(h[0][1] == 1){
+                      // inner corner at front
+                      cX.at(0) = 0;
+                      cY.at(0) = -1;
+                      cX.resize(2,(int)-1);
+                      cY.resize(2,(int)0);
```

Comment:
My bad! `cx, cy` change in size here. 

[std::vector::pushback](http://www.cplusplus.com/reference/vector/vector/push_back/) instead of resize. pushback is like list-append in python.